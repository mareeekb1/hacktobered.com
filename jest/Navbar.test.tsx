import { Backdrop, MobileNav, Transition } from "@/components/NavBar/MobileNav";
import { DesktopNavLink, MobileNavLink } from "@/components/NavBar/NavLink";
import { Button } from "@chakra-ui/react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

describe("Mobile navigation bar check rendering", () => {
  test("Backdrop test", () => {
    const component = renderer.create(<Backdrop />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Transition test", () => {
    const component = renderer.create(<Transition />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Full mobile navigation test", () => {
    const component = renderer.create(<MobileNav />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Backdrop hide", () => {
    const box = renderer.create(<Backdrop />).toTree()?.rendered.props.children
      .props.animate;
    expect(box).toBe("hide");
  });
  test("Backdrop show", () => {
    const box = renderer.create(<Backdrop show />).toTree()?.rendered.props
      .children.props.animate;
    expect(box).toBe("show");
  });
});

describe("Desktop navigation test", () => {
  test("DesktopNavLinku test", () => {
    const component = renderer.create(<DesktopNavLink />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("DesktopNavLink show", () => {
    const box = renderer.create(<DesktopNavLink active />).toTree()?.rendered;
    const props = box[1].props["aria-current"];
    expect(props).toBe("page");
  });
  test("DesktopNavLink show", () => {
    const box = renderer.create(<DesktopNavLink />).toTree()?.rendered;
    const props = box[1].props["aria-current"];
    expect(props).toBe(undefined);
  });
  test("MobileNavLink test", () => {
    const mockFn = jest.fn();
    const component = renderer.create(
      <MobileNavLink icon="abbr">
        <Button onClick={mockFn}>test</Button>
      </MobileNavLink>
    );
    screen;
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("MobileNavLink with mocked button test", async () => {
    const mockFn = jest.fn();
    const component = render(
      <MobileNavLink icon="abbr">
        <Button onClick={mockFn}>test</Button>
      </MobileNavLink>
    );
    const btn = await screen.findByRole("button");
    btn.click();
    expect(mockFn.mock.calls.length).toBe(1);
  });
});
