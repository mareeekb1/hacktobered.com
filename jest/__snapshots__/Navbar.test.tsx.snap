// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop navigation test DesktopNavLinku test 1`] = `
<a
  className="css-72zd6n"
  href="#"
/>
`;

exports[`Desktop navigation test MobileNavLink test 1`] = `
<a
  className="css-1qw5rjk"
>
  <svg
    className="chakra-icon css-5084o3"
    focusable={false}
    viewBox="0 0 24 24"
  >
    <g
      stroke="currentColor"
      strokeWidth="1.5"
    >
      <path
        d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
        fill="none"
        strokeLinecap="round"
      />
      <path
        d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
        fill="currentColor"
        strokeLinecap="round"
      />
      <circle
        cx="12"
        cy="12"
        fill="none"
        r="11.25"
        strokeMiterlimit="10"
      />
    </g>
  </svg>
  <div
    className="css-nh8546"
  >
    <button
      className="chakra-button css-4xx2wk"
      onClick={[MockFunction]}
      type="button"
    >
      test
    </button>
  </div>
</a>
`;

exports[`Mobile navigation bar check rendering Backdrop test 1`] = `<span />`;

exports[`Mobile navigation bar check rendering Full mobile navigation test 1`] = `
[
  <button
    className="css-k42n6e"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 6h16M4 12h16M4 18h16"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>,
  <div
    style={
      {
        "display": "none",
        "left": "16px",
        "margin": "0 auto",
        "opacity": 0,
        "position": "absolute",
        "top": "24px",
        "transform": "scale(0.98) translateZ(0)",
        "transformOrigin": "top right",
        "width": "calc(100% - 32px)",
        "zIndex": 1,
      }
    }
  >
    <div
      onScrollCapture={[Function]}
      onTouchMoveCapture={[Function]}
      onWheelCapture={[Function]}
    >
      <span />
    </div>
    <div
      data-focus-guard={true}
      style={
        {
          "height": "0px",
          "left": "1px",
          "overflow": "hidden",
          "padding": 0,
          "position": "fixed",
          "top": "1px",
          "width": "1px",
        }
      }
      tabIndex={-1}
    />
    <div
      data-focus-lock-disabled="disabled"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="css-b0ao7w"
        tabIndex={0}
      >
        <div
          className="css-1jgcock"
        >
          <div
            className="css-1lekzkb"
          >
            <span
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              Hacktobered
            </span>
            <div
              className="css-e4v5av"
            >
              <button
                className="css-i12i7u"
                onClick={[Function]}
                type="button"
              >
                <div
                  className="css-8b45rq"
                >
                  Close menu
                </div>
                <svg
                  aria-hidden={true}
                  fill="none"
                  fontSize="1.5rem"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 18L18 6M6 6l12 12"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                  />
                </svg>
              </button>
            </div>
          </div>
          <nav
            className="css-422jt4"
          >
            <a
              className="css-1qw5rjk"
              href="/"
            >
              <svg
                className="chakra-icon css-597cn"
                fill="currentColor"
                focusable={false}
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 4.804A7.968 7.968 0 005.5 4c-1.255 0-2.443.29-3.5.804v10A7.969 7.969 0 015.5 14c1.669 0 3.218.51 4.5 1.385A7.962 7.962 0 0114.5 14c1.255 0 2.443.29 3.5.804v-10A7.968 7.968 0 0014.5 4c-1.255 0-2.443.29-3.5.804V12a1 1 0 11-2 0V4.804z"
                />
              </svg>
              <div
                className="css-nh8546"
              >
                Hacktobered
              </div>
            </a>
            <a
              className="css-1qw5rjk"
              onClick={[Function]}
            >
              <svg
                className="chakra-icon css-597cn"
                fill="none"
                focusable={false}
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                />
              </svg>
              <div
                className="css-nh8546"
              >
                Sign Out
              </div>
            </a>
          </nav>
        </div>
      </div>
    </div>
    <div
      data-focus-guard={true}
      style={
        {
          "height": "0px",
          "left": "1px",
          "overflow": "hidden",
          "padding": 0,
          "position": "fixed",
          "top": "1px",
          "width": "1px",
        }
      }
      tabIndex={-1}
    />
  </div>,
]
`;

exports[`Mobile navigation bar check rendering Transition test 1`] = `
<div
  style={
    {
      "display": "none",
      "left": "16px",
      "margin": "0 auto",
      "opacity": 0,
      "position": "absolute",
      "top": "24px",
      "transform": "scale(0.98) translateZ(0)",
      "transformOrigin": "top right",
      "width": "calc(100% - 32px)",
      "zIndex": 1,
    }
  }
/>
`;
